environment/container/kubernetes/k3s.go

```diff
@@ -156,18 +157,24 @@ func installK3sCluster(
 		"--write-kubeconfig-mode", "644",
 	}, k3sArgs...)
 
-	// replace ip address if networking is enabled
-	ipAddress := limautil.IPAddress(config.CurrentProfile().ID)
-	if ipAddress == "127.0.0.1" {
-		args = append(args, "--flannel-iface", "eth0")
-	} else {
-		if !hasK3sArg(k3sArgs, "--advertise-address") {
-			args = append(args, "--advertise-address", ipAddress)
+	a.Retry("waiting for VM IP address", time.Second*2, 10, func(retryCount int) error {
+		ipAddress := limautil.IPAddress(config.CurrentProfile().ID)
```

Comment:
Something I'm not sure about is whether the actual network interface is guaranteed to exist at this point. Otherwise this will just end up falling back to 127.0.0.1. That behavior is however unchanged by this.